webviz_subsurface/plugins/_bhp_qc/_plugin.py

```diff
@@ -0,0 +1,204 @@
+from pathlib import Path
+from typing import Dict, List, Type
+
+import pandas as pd
+from dash.development.base_component import Component
+from webviz_config import WebvizPluginABC, WebvizSettings
+
+from webviz_subsurface._providers import EnsembleSummaryProvider
+
+from .._simulation_time_series._utils.create_provider_set_from_paths import (
+    create_lazy_provider_set_from_paths,
+)
+from ._error import error
+from ._plugin_ids import PluginIds
+from .shared_settings import BarLineSettings, Filter
+from .view_elements import Graph
+from .views import BarView, FanView, LineView
+
+
+class BhpQc(WebvizPluginABC):
+    """QC simulated bottom hole pressures (BHP) from reservoir simulations.
+
+    Can be used to check if your simulated BHPs are in a realistic range.
+    E.g. check if your simulated bottom hole pressures are very low in producers,
+    or very high injectors.
+    ---
+
+    * **`ensembles`:** Which ensembles in `shared_settings` to visualize.
+    * **`rel_file_pattern`:** path to `.arrow` files with summary data.
+    ---
+    Data is read directly from the arrow files with the raw frequency (not resampled).
+    Resampling and csvs are not supported to avoid potential of interpolation, which
+    might cover extreme BHP values.
+
+    """
+
+    def __init__(
+        self,
+        webviz_settings: WebvizSettings,
+        ensembles: list,
+        rel_file_pattern: str = "share/results/unsmry/*.arrow",
+        wells: List[str] = None,
+    ):
+        super().__init__()
+
+        self.error_message = ""
+
+        if ensembles is None:
+            self.error_message = "Enembles needs to be provided"
+            raise ValueError("Enembles needs to be provided")
+
+        self.ens_paths: Dict[str, Path] = {
+            ensemble_name: webviz_settings.shared_settings["scratch_ensembles"][
+                ensemble_name
+            ]
+            for ensemble_name in ensembles
+        }
+
+        try:
+            self._input_provider_set = create_lazy_provider_set_from_paths(
+                self.ens_paths,
+                rel_file_pattern,
+            )
+        except PermissionError:
+            self.error_message = f"Access to files '{rel_file_pattern}' denied."
+            "Please check your path for 'rel_file_pattern'"
+            return
+        except FileNotFoundError:
+            self.error_message = f"Files '{rel_file_pattern}' not found."
+            "Please check your path for 'rel_file_pattern'."
+            return
```

Comment:
Why do we not want to fail in these two cases? 🙂

Reply:
Removed the exceptions all-together as they are already handled implicitly by `create_lazy_provider_set_from_paths`